client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -116,6 +117,28 @@ const OnboardingEdit: React.FunctionComponent< BlockEditProps< Attributes > > =
 		return redirectToLatestStep;
 	}
 
+	// Remember the last accessed route path
+	const location = useLocation();
+	const step = useStepRouteParam();
+	const { setLastLocation } = useDispatch( STORE_KEY );
+
+	React.useEffect( () => {
+		if (
+			// When location.key is undefined, this means user has just entered gutenboarding from url.
+			location.key !== undefined &&
+			// When step exists, and step is not any from the modals
+			step &&
+			! [
+				Step.DomainsModal as string,
+				Step.PlansModal as string,
+				Step.LanguageModal as string,
+			].includes( step )
+		) {
+			// Remember last location
+			setLastLocation( location.pathname );
+		}
+	}, [ location, step, setLastLocation ] );
```

Comment:
I would do this for readability and to avoid casting.

```suggestion
	React.useEffect( () => {
		const modalSteps: StepType[] = [ Step.DomainsModal, Step.PlansModal, Step.LanguageModal ];
		if (
			// When location.key is undefined, this means user has just entered gutenboarding from url.
			location.key !== undefined &&
			// When step exists, and step is not any from the modals
			step &&
			! modalSteps.includes( step )
		) {
			// Remember last location
			setLastLocation( location.pathname );
		}
	}, [ location, step, setLastLocation ] );
```

Reply:
Thanks. I was scratching my head how to fix this!